agenta-cli/agenta/cli/main.py

```diff
@@ -87,99 +88,101 @@ def init(app_name: str):
                         "Invalid input. Please use only alphanumeric characters without spaces."
                     )
 
-    where_question = questionary.select(
-        "Where are you running agenta?",
-        choices=["On my local machine", "On a remote machine", "On agenta cloud"],
-    ).ask()
-
-    if where_question == "On my local machine":
-        backend_host = "http://localhost"
-    elif where_question == "On a remote machine":
-        backend_host = questionary.text(
-            "Please provide the IP or URL of your remote host"
+    try:
+        where_question = questionary.select(
+            "Where are you running agenta?",
+            choices=["On my local machine", "On a remote machine", "On agenta cloud"],
         ).ask()
-    elif where_question == "On agenta cloud":
-        backend_host = "https://demo.agenta.ai"
 
-        api_key = questionary.text("Please provide your API key:").ask()
+        if where_question == "On my local machine":
+            backend_host = "http://localhost"
+        elif where_question == "On a remote machine":
+            backend_host = questionary.text(
+                "Please provide the IP or URL of your remote host"
+            ).ask()
+        elif where_question == "On agenta cloud":
+            backend_host = "https://demo.agenta.ai"
+
+            api_key = helper.get_api_key()
+            client.validate_api_key(api_key, backend_host)
```

Comment:
What happens if validate_api_key fails? It seems we throw an exception. In that case we should catch it and provide some helpful message to the user